### PR TITLE
Improve kits and kit parsing

### PR DESCRIPTION
**Pull request by thomblr**
Some updates about kits :
- No longer using NMS for unbreakable attribute.
- Merge armor item parsing into `Parser.getItem()` method.
- Add Enderpearl reset
- Add KitFly into kit

Still WIP, I'll squash commits when it'll be finished.

### CODE REVIEW

**Comment by CaptainElliott** [comment]
I prefer the event name `KitApplyEvent`

File: src/main/java/in/twizmwaz/cardinal/event/KitApplyEvent.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    public KitApplyEvent(Kit kit, Player player) {
+        this.kit = kit;
+        this.player = player;
+    }
+
+    public static HandlerList getHandlerList() {
+        return handlers;
+    }
+
+    public HandlerList getHandlers() {
```
Comment (twizmwazin) [inline src/main/java/in/twizmwaz/cardinal/event/KitApplyEvent.java]: `@Override`

**Comment by twizmwazin** [comment]
This looks pretty good so far, although I believe there are a handful of unused imports you can remove.

**Comment by thomblr** [comment]
@twizmwazin I had already cleaned imports but forgot one. Tell me when you agree the changes and I squash it.

**Comment by CaptainElliott** [comment]
You should probably clean up with PR, twiz doesn't usually like more than one commit ;) :p

**Comment by thomblr** [comment]
@CaptainElliott That's why I'll squash the commits when twiz agrees my changes because I want to do this only once.

**Comment by thomblr** [comment]
@twizmwazin I squashed the commits, I let you doing what you want with these changes now.

**Comment by dentmaged** [comment]
@thomblr how did you do that?

**Comment by thomblr** [comment]
@dentmaged 
- `git rebase -i HEAD~{numberOfCommitBehind}`
- change `pick` to `squash` in the shown editor on which commits you want to merge together, don't forget to let one with "pick"
- save with :wq
- `push --force origin {yourBranch}`

**Comment by dentmaged** [comment]
@thomblr Awesome! Thanks for letting me know!

**Comment by thomblr** [comment]
@twizmwazin Fixed

**Comment by twizmwazin** [comment]
If we can get some feedback I think this can merged soon

**Comment by Alfalik** [comment]
This kit is not being applied to the players and it's throwing an IllegalArgumentException (http://pastebin.com/Y4yKVq9C).

```xml
<teams>
	<team id="blue" color="blue" max="10">Blue</team>
	<team id="red" color="dark red" max="10">Red</team>
</teams>
<spawns>
	<spawn team="blue" kit="blue" yaw="-90">
		<cylinder base="-178,27,0" radius="2" height="0"/>
	</spawn>
	<spawn team="red" kit="red" yaw="90">
		<cylinder base="-18,27,0" radius="2" height="0"/>
	</spawn>
	<default yaw="0">
		<cylinder base="-98,27,0" radius="2" height="0"/>
	</default>
</spawns>
<kits>
	<kit name="spawn">
		<item slot="0">stone sword</item>
		<item slot="1">bow</item>
		<item slot="2" amount="16">cooked chicken</item>
		<item slot="3" amount="16">leaves</item>
		<item slot="8" amount="8">arrow</item>
	</kit>
	<kit name="red" parents="spawn">
		<helmet color="cd0000">leather helmet</helmet>
		<chestplate color="cd0000">leather chestplate</chestplate>
		<leggings color="cd0000">leather leggings</leggings>
		<boots color="cd0000">leather boots</boots>
	</kit>
	<kit name="blue" parents="spawn">
		<helmet color="0066cc">leather helmet</helmet>
		<chestplate color="0066cc">leather chestplate</chestplate>
		<leggings color="0066cc">leather leggings</leggings>
		<boots color="0066cc">leather boots</boots>
	</kit>
	<kit name="red-flag-kit">
		<potion duration="oo">slowness</potion>
		<potion duration="oo">damage_resistance</potion>
	</kit>
	<kit name="blue-flag-kit">
		<potion duration="oo">slowness</potion>
		<potion duration="oo">damage_resistance</potion>
	</kit>
	<kit name="reset-kit" force="true">
		<potion duration="0" amplifier="10">slowness</potion>
		<potion duration="0" amplifier="10">damage_resistance</potion>
	</kit>
</kits>
```
Here's another map that kits are not being applied (XML: http://pastebin.com/J80SHFee, Log: http://pastebin.com/1HMuY2Mq).
The rest is working as it should.

**Comment by thomblr** [comment]
@Alfalik I tried to fix it by using Minecraft potion names too because potion name "slowness" is only in minecraft potion names. Can you try again and tell me if it's working ?

**Comment by Alfalik** [comment]
@thomblr Yep. It fixed it, good job. Hope this get merged soon, it's working as it should.

File: src/main/java/in/twizmwaz/cardinal/module/modules/enderpearlReset/EnderpearlReset.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                Player shooter = (Player) event.getEntity().getShooter();
+                thrownPearls.put(shooter.getUniqueId(), event.getEntity());
+            }
+        }
+    }
+
+    @EventHandler
+    public void onApplyingKit(KitApplyEvent event) {
+        if (event.getKit().isResetPearls()) {
+            Iterator<Map.Entry<UUID, Entity>> iterator = thrownPearls.entrySet().iterator();
```
Comment (ryanw-se) [inline src/main/java/in/twizmwaz/cardinal/module/modules/enderpearlReset/EnderpearlReset.java]: I would replace this style of a iterator with something that is more conventional, something along the lines of this:

```java
for (Map.Entry<UUID, Entity> entry : thrownPearls.entrySet()) {
    if (entry.getKey().equals(event.getPlayer().getUniqueId())) {
        if (!entry.getValue().isDead()) {
            entry.getValue().remove();
        }
    }
}
```
}

**Comment by Pablete1234** [comment]
@thomblr is the reset enderpearl module needed? i feel like on the applying kit part you could just run an:
```
if (Kit.isResetPearls()){
  for (Entity entity : get world entity list){
    if (Entity equals enderpearl  && projectile thrower equals player){
     kill enderpearl
    }
  }
}
```

**Comment by Pablete1234** [comment]
I guess this was implemented in my kits rewrite

**Pull request closed without merging**